microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/model/PropertyValueExpression.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.lsp4mp.model;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a portion of the property value that refers to the value of
+ * another property.
+ *
+ * When properties file is processed, the reference is replaced with the value
+ * of the other property. In the properties file, it has the form:
+ * <code>${other.property.name}</code>
+ */
+public class PropertyValueExpression extends Node {
+
+	private static Pattern PROPERTY_NAME_EXTRACTOR = Pattern.compile("\\$\\{([^\\}]*)\\}?");
+
+	@Override
+	public NodeType getNodeType() {
+		return NodeType.PROPERTY_VALUE_EXPRESSION;
+	}
+
+	/**
+	 * Returns the text that this Node contains.
+	 *
+	 * Removes backslashes, and newlines. Doesn't not resolve the reference to
+	 * another property.
+	 */
+	public String getValue() {
+		String text = getText(true);
+		return text != null ? text.trim() : null;
+	}
+
+	/**
+	 * Get the name of the referenced property, or null if brackets were missing in
+	 * the property expression.
+	 *
+	 * Does not check if the referenced property exists.
+	 *
+	 * @return the name of the referenced property, or null if brackets were missing
+	 *         in the property expression.
+	 */
+	public String getReferencedPropertyName() {
+		Matcher m = PROPERTY_NAME_EXTRACTOR.matcher(getValue());
```

Comment:
Can we avoid using Regexp here just to extract property name? Using Regexp a lot, can grow up the memory or provide bad performance. @fbricon are you OK with that?